service/controller/v4/resource/chart/desired.go

```diff
@@ -101,6 +147,40 @@ func (r *Resource) getSecretValues(ctx context.Context, customObject v1alpha1.Ch
 	return secretValues, nil
 }
 
+// mergeValuesConfigMaps merges values generated by the controlling operator
+// with userValues overriden by the user. Only values under the configmap key
+// can be overriden.
+func mergeValuesConfigMaps(values, userValues map[string]interface{}) (map[string]interface{}, error) {
+	key := "configmap"
+
+	if values == nil || len(values) == 0 {
+		return userValues, nil
+	}
+	if userValues == nil || len(userValues) == 0 {
+		return values, nil
+	}
+
+	raw, ok := values[key]
+	if !ok {
+		values[key] = userValues
+		return values, nil
+	}
+
+	vals, ok := raw.(map[string]interface{})
```

Comment:
Does it somehow matter that the original value is modified? Should we provide some copy support?

Reply:
We want the user values to override the values generated by the operator. But it might be safer to copy the settings into a new map. I'll take another look at it.

Reply:
Merge logic is updated. I kept the approach of modifying the generated values. The code is cleaner that way and the merge logic is still safe IMO.